.htaccess

```
@@ -138,53 +141,71 @@ Redirect "/products/keymanweb" "/products/web"
 
 Redirect "/bookmarklet" "/products/bookmarklet"
 
+#
+# Keyboard redirects
+#
+
+# Redirect broken legacy keyboard links
+RewriteRule "^keyboard/gff_(mdx|muz|suq)_7(/.*)?" "keyboard/gff_mym_7$2" [R,L]
+
+RewriteRule "^keyboard/redir.php(.*)$" - [END]
+
+# Rewrite /keyboard/[name][/] to /keyboard/redir.php?q=[name]
+# only if index.php does not exist in the folder
+RewriteCond "%{DOCUMENT_ROOT}/keyboard/$1/index.php" !-f
+RewriteRule "^keyboard/(?!redir.php)([^/?]+)(/)?$" "keyboard/redir.php?q=$1" [END]
+
+# Rewrite /keyboard/[name]/[version][/] to /keyboard/redir.php?q=[name]&amp;v=[version]
+# only if index.php does not exist in the folder
+RewriteCond "%{DOCUMENT_ROOT}/keyboard/$1/$2/index.php" !-f
+RewriteRule "^keyboard/(?!redir.php)([^/?]+)/([^/?]+)(/)?$" "keyboard/redir.php?q=$1&v=$2" [END]
+
+
 #
 # PHP and Markdown rewriting
 #
 
 # special case because of specific filenames : these don't redirect properly otherwise
 
 # Rewrite nul -> _nul, index -> _index
-RewriteRule "^developer/language/reference/(nul|index)(\.php)?$" "developer/language/reference/_$1.php" [L]
+RewriteRule "^developer/language/reference/(nul|index)(\.php)?$" "developer/language/reference/_$1.php" [END]
+
+# no pages currently reference this but will allow for future automatic re-versioning links in docs
+RewriteRule "^developer/15.0/guides/develop/creating-a-touch-keyboard-layout-for-amharic-with-keyman-developer-10(.*)?$" "developer/15.0/guides/develop/creating-a-touch-keyboard-layout-for-amharic.php" [END]
 
 # Remove index or index.php and redirect (and stop processing)
-RewriteCond "$1/index" -f [OR]
-RewriteCond "$1/index.php" -f
-RewriteRule "^((.+)/)?index(\.php)?$" "/$1" [R,L]
+RewriteCond "{DOCUMENT_ROOT}/$1" -d
+RewriteRule "^((.+)/)?index(\.php)?$" "$1" [R,L]
 
 # Remove .php extension and redirect
-RewriteCond "$1.php" -f
+RewriteCond "{DOCUMENT_ROOT}/$1.php" -f
```

```suggestion
RewriteCond "$1.php" -f
```

> As @ermshiperete says, I think this should be restored to the original.

I tried the suggestion, but the keyboard pages failed to load

---

nvm, it's working now without `{DOCUMENT_ROOT}`